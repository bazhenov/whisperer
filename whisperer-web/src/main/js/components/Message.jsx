import React, { Component } from 'react'
import moment from 'moment'
import MessageArgument from './MessageArgument.jsx';
import { SQUASHED_MESSAGE_ARGUMENT_LENGTH } from '../constants'

export default class Message extends Component {

	shouldComponentUpdate() {
		return false;
	}

	render() {
		const { message } = this.props;
		const args = message.args || [];
		const textParts = message.message.split('{}').reduce((values, part, i) => {
			values.push(<span key={'p' + i}>{part}</span>);
			if (i < args.length) {
				const arg = String(args[i]);
				const key = 'a' + i;
				const argumentEl = arg.length <= SQUASHED_MESSAGE_ARGUMENT_LENGTH ? <span key={key}>{arg}</span> :
					<MessageArgument key={key} text={arg} />;
				values.push(argumentEl);
			}
			return values;
		}, []);

		return <div className="message-row">
			<div className="message-head">
				<div className="message-time">{moment(message.timestamp).format("HH:mm:ss.SSS")}</div>
				<div className="message-host">@{message.host}</div>
				<div className="message-group">{message.shortGroupName}</div>
				<div className="message-thread">{message.thread}</div>
			</div>
			<div className="message-content">{ textParts }</div>
		</div>
	}
}