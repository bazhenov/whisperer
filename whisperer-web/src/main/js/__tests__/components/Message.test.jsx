import React from "react";
import Message from "../../components/Message.jsx";
import renderer from "react-test-renderer";
import { message } from '../fixtures'

it('Message', () => {
	expect(renderer.create(
		<Message message={message} />
	).toJSON()).toMatchSnapshot()
});