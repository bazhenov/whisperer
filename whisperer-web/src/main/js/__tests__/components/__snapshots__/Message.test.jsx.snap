// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message 1`] = `
<div
  className="message-row"
>
  <div
    className="message-head"
  >
    <div
      className="message-time"
    >
      14:17:09.683
    </div>
    <div
      className="message-host"
    >
      @
      http://search-coord5.akod.loc
    </div>
    <div
      className="message-group"
    >
      c.f.s.CoordinatorSearchEngine
    </div>
    <div
      className="message-thread"
    >
      qtp-12345
    </div>
  </div>
  <div
    className="message-content"
  >
    <span>
      Delegating execution to command 
    </span>
    <span>
      StagedSearchQuery
    </span>
    <span>
      , query: '
    </span>
    <a
      href="#"
      onClick={[Function]}
    >
      ftext contains('s
      ...
    </a>
    <span>
      ', sort: 
    </span>
    <span>
      null
    </span>
    <span>
      , facets: 
    </span>
    <span>
      directoryId
    </span>
    <span>
      , filter: '
    </span>
    <a
      href="#"
      onClick={[Function]}
    >
      siteId!='1632' an
      ...
    </a>
    <span>
      '
    </span>
  </div>
</div>
`;
